pkg/clusteragent/admission/mutate/agent_sidecar/README.md

```diff
@@ -0,0 +1,87 @@
+- Enabling the feature
+- Setting custom selectors
+- Setting custom profiles
+- Setting a provider
+- Creating datadog secrets in every namespace where you wish to inject the agent sidecar
+
+This allows the user to customize the matching criteria for the webhook. It allows specifying which pods will be injected with the agent sidecar.
+
+With this configuration, the default agent sidecar will be injected, in addition to any overrides set by the user.
+
+Having set a provider, the agent sidecar will also get automatically the necessary configurations for the targeted provider.
+
+## Expected Behaviour
+
+The table below shows the expected behaviour when the feature is enabled.
+
+Note that currently we only support creating 1 selector and 1 profile (config override).
+Creating multiple selectors and/or overrides will result in not registering any webhook.
+
+| Custom Selectors / Profiles Set | Provider Set       | Provider Supported | Expected Behaviour                                                                                                                                      |
+|---------------------------------|--------------------|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------|
+| :x:                             | :heavy_check_mark: | :heavy_check_mark: | Agent sidecar should be injected on pods having the provider label key set (`agent.datadoghq.com/sidecar: <provider>`)                                  |
+| :x:                             | :heavy_check_mark: | :x:                | No agent sidecar should be injected, and an error message will be logged in the cluster agent:  "agent sidecar provider is not supported: foo-provider" |
+| :x:                             | :x:                | :x:                | No agent sidecar should be injected, and an error message will be logged in the cluster agent:  "agent sidecar provider is not supported"               |
+| :heavy_check_mark:              | :heavy_check_mark: | :heavy_check_mark: | The agent sidecar container should be injected only on pods matching the selector, and the `DD_EKS_FARGATE` label should be set to `true`               |
+| :heavy_check_mark:              | :heavy_check_mark: | :x:                | Agent sidecar should be injected, and you must find an error message in the cluster agent logs "unsupported provider: foo"                              |
+| :heavy_check_mark:              | :x:                | :x:                | The agent sidecar container should be injected only on pods matching the selector                                                                       |
+
+
+
+
+## Notes
+- For now, we only support configuring 1 custom selector and 1 custom profile.
+- For now, only `fargate` provider is supported
+- For now, only 1 selector and 1 profile (config override) can be configured.
```

Comment:
Might be useful to add an example of how to define the custom selector.

Reply:
Yes indeed.

Currently we still didn't add configs for this feature in helm and operator. Showing an example here would require configuring environment variables (as you can see in the QA), which is not very nice to have.

I think it would be better to wait until @levan-m updates helm and the operator to support configuring this feature easily and then we can update the readme here.